### ISSUE TITLE
Apply with identity and mask doesn't preserve iso-valued-ness

### ISSUE THREAD

**Issue by eriknw** [DrTimothyAldenDavis/GraphBLAS#70]
For example:
```C
  GrB_Matrix_new(&M_0, GrB_INT64, 4, 4);
  GrB_Matrix_new(&M_1, GrB_INT64, 4, 4);
  GrB_Matrix_assign_INT64(M_0, NULL, NULL, 1, GrB_ALL, 4, (GrB_Index[]){0, 1}, 2, NULL);
  GrB_Matrix_assign_INT64(M_1, NULL, NULL, 1, GrB_ALL, 4, (GrB_Index[]){1, 2}, 2, NULL);
  GrB_Matrix_new(&M_2, GrB_INT64, 4, 4);
  GrB_Matrix_apply(M_2, M_1, NULL, GrB_IDENTITY_INT64, M_0, GrB_DESC_S);
```
`M_0` and `M_1` are both iso-valued with the same datatype and value of `1`.  I expect `M_2` to be iso-valued, but it is not.

For this example, I can use assign instead of apply, which would preserve iso-valued-ness.

**Comment by DrTimothyAldenDavis** [comment]
Oops.  I think the problem is in GB_masker, which does C<M>=Z for many methods.  I have a check, called GB_iso_masker (in GB_masker_phase2) that determines if C is iso.  It's iso if C and Z are both iso and have the same iso value.

In this case, the matrix Z returned by the internal apply is iso.  But M_2 (the C matrix in C<M>=Z) is empty.  An empty matrix is not iso, so I determine that the resulting C matrix is no longer iso.

That's a mistake.  My test in GB_iso_masker should check if C is empty, in which case C is iso if Z is iso, and the check for C and Z having the same iso value should be skipped.

I think this is an easy fix.  It affects not only GrB_apply, but all operations that use the mask/accum phase via GB_masker (all operations do that like GrB_mxm, etc).  Thanks for catching this.

**Comment by DrTimothyAldenDavis** [comment]
This bug doesn't affect GrB_assign or GxB_subassign, since those methods do their own masking.  They don't use GB_masker.

**Comment by DrTimothyAldenDavis** [comment]
Fixed in 5.2.0.alpha9.
